src/main/java/MultiUsecaseUtil/SerializeDataAccess.java

```diff
@@ -0,0 +1,185 @@
+        try {
+            output = new ObjectOutputStream(buffer);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        Group newGroup = groupDSRequestModel.getGroup();
+        this.groupMap.put(newGroup.getGroupName(), newGroup);
+        try {
+            output.writeObject(this.groupMap);
+            output.close();
+            buffer.close();
+            file.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public HashMap<String, Group> loadGroups(){
+        InputStream file = null;
+        try {
+            file = new FileInputStream("database/group.ser");
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+        InputStream buffer = new BufferedInputStream(file);
+        ObjectInput input = null;
+        try {
+            input = new ObjectInputStream(buffer);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        HashMap<String, Group> groups;
+
+        Object ReadFromInput = null;
+        try {
+            ReadFromInput = input.readObject();
+        } catch (ClassNotFoundException | IOException e) {
+            throw new RuntimeException(e);
+        }
+        if (ReadFromInput.equals("")){
+            groups = new HashMap<String, Group>();
+        }
+        else{
+            groups = (HashMap<String, Group>) ReadFromInput;
+        }
+        try {
+            input.close();
+            buffer.close();
+            file.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return groups;
+    }
+
+    @Override
+    public boolean groupIdentifierExists(String groupName) {
+        try{
+            return this.groupMap.containsKey(groupName);
+        } catch (RuntimeException e){
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void saveNewUser(UserRegistrationDSRequestPackage userDSRequestModel){
+        OutputStream file = null;
+        try {
+            file = new FileOutputStream("database/user.ser");
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+        OutputStream buffer = new BufferedOutputStream(file);
+        ObjectOutput output = null;
+        try {
+            output = new ObjectOutputStream(buffer);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        User newUser = userDSRequestModel.getUser();
+        this.userMap.put(newUser.getUsername(), newUser);
+        try {
+            output.writeObject(this.userMap);
+            output.close();
+            buffer.close();
+            file.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public HashMap<String, User> loadUsers(){
```

Comment:
To get the current User by username, would we loadUsers() and then search for the user in the Hashmap? Or are we still going to implement a loadUser(String username)?

Reply:
implement a loaduser in your dsgateway interface